File: c++_code/ThreadExample.cpp
#include <iostream>
#include <thread>

void printFromThread() {
    std::cout << "Hello from thread" << std::endl;
}

int main() {
    // Create a thread that runs the function printFromThread
    std::thread t(printFromThread);

    // Print from the main thread
    std::cout << "Hello from Main" << std::endl;

    // Wait for the thread to complete
    t.join();

    return 0;
}

File: python_code/thread_example.py
import threading

def print_from_thread():
    print("Hello from thread")

# Create a new thread that runs the print_from_thread function
thread = threading.Thread(target=print_from_thread)
thread.start()

# Print from the main thread
print("Hello from Main")

# Wait for the thread to complete
thread.join()